lib/google/serverless/exec/tasks.rb

```diff
@@ -297,7 +300,7 @@ def verify_gcloud_and_report_errors
           end
     
           def start_and_report_errors app_exec
-            app_exec.start
+            app_exec.product == "app_engine" ? app_exec.start_app_engine : app_exec.start_cloud_run
```

Comment:
Consider retaining the `app_exec.start` method, and having _that method_ check the product and branch off either to `start_app_engine` or `start_cloud_run`. That way, the caller doesn't have to know about product.

It's good practice when designing classes to limit what the user of a class needs to know in order to use the class, and limit the ways in which the caller could make a mistake. In this case, it would be an error to call `start_cloud_run` if the product was app engine. So just have the Exec class itself handle that logic, and remove even the possibility of the caller making that mistake.